### PR TITLE
Add doc of sinking data in parquet format

### LINKED ISSUE #2497: Document: feat(sink): introduce file sink in PARQUET format
This issue tracks the documentation update needed for the merged PR #17311.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/17311
Source PR Merged At: 2024-08-22T02:39:08Z

If it is a major improvement that deserves a new page or a new section in the documentation, please check if we should label it as an experiment feature.

### PR DESCRIPTION
**Pull request by WanYixian**
## Description

Add doc of sinking data in parquet format, 
@wcy-fdu let me know if any other info is worth mentioning in the doc, I'll add them accordingly, thanks!

- [x] update public preview list

## Related code PR

https://github.com/risingwavelabs/risingwave/pull/17311

## Related doc issue

Resolve https://github.com/risingwavelabs/risingwave-docs/issues/2497

## Checklist

- [ ] I have checked the doc site preview, and the updated parts look good.
- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

File: docs/data-delivery.md [hunk 1]
```diff
@@ -160,3 +160,34 @@ When creating an `upsert` sink, note whether or not you need to specify the prim
...
+
+RisingWave supports sinking data in Parquet format to file systems including S3, Google Cloud Storage (GCS), and Azure Blob Storage. This eliminates the need for complex data lake setups. Once the data is saved, the files can be queried using the batch processing engine of RisingWave through the `file_scan` API. You can also leverage third-party OLAP query engines for further data processing.
+
+Below is an example to sink data to S3:
+
+```sql
+CREATE SINK test_file_sink FROM test 
+WITH (
+    connector = 's3',
+    match_pattern = '*.parquet',
```
Comment (wcy-fdu) [inline docs/data-delivery.md]: `create sink` statement does not have a `match_patten` field, it only targets source.

File: docs/data-delivery.md [hunk 2]
```diff
@@ -160,3 +160,34 @@ When creating an `upsert` sink, note whether or not you need to specify the prim
...
+    s3.region_name = '{config['S3_REGION']}',
+    s3.bucket_name = '{config['S3_BUCKET']}',
+    s3.credentials.access = '{config['S3_ACCESS_KEY']}',
+    s3.credentials.secret = '{config['S3_SECRET_KEY']}',
+    s3.endpoint_url = 'https://{config['S3_ENDPOINT']}'
+    s3.path = '',
+    s3.file_type = 'parquet',
```
Comment (wcy-fdu) [inline docs/data-delivery.md]: `s3.file_type = 'parquet',` no need this line, the encode type is defined in `FORMAT ... ENCODE ...`.

File: docs/data-delivery.md [hunk 3]
```diff
@@ -160,3 +160,32 @@ When creating an `upsert` sink, note whether or not you need to specify the prim
...
+    connector = 's3',
+    s3.region_name = '{config['S3_REGION']}',
+    s3.bucket_name = '{config['S3_BUCKET']}',
+    s3.credentials.access = '{config['S3_ACCESS_KEY']}',
+    s3.credentials.secret = '{config['S3_SECRET_KEY']}',
+    s3.endpoint_url = 'https://{config['S3_ENDPOINT']}'
+    s3.path = '',
+    type = 'append-only',
+    force_append_only='true'
+) FORMAT PLAIN ENCODE PARQUET(force_append_only='true')
```
Comment (ShanlanLi) [inline docs/data-delivery.md]:
```suggestion
) FORMAT PLAIN ENCODE PARQUET(force_append_only='true');  
```

**Review by ShanlanLi — APPROVED** [review]
LGTM

**Pull request merged**